src/pages/Encounters/tabs/responses.tsx

```diff
@@ -165,17 +171,32 @@ export const EncounterResponsesTab = ({
 }: EncounterResponsesTabProps) => {
   const { t } = useTranslation();
   const [qParams, setQueryParams] = useQueryParams<{
+    questionnaireSlug?: string;
     questionnaireId?: string;
     responseId?: string;
   }>();
 
-  const { questionnaireId, responseId } = qParams;
+  const { questionnaireSlug, questionnaireId, responseId } = qParams;
 
   const [selectedQuestionnaireTitle, setSelectedQuestionnaireTitle] =
     useState<string>("");
   const [isDrawerOpen, setIsDrawerOpen] = useState(false);
   const isMobile = useBreakpoints({ default: true, md: false });
 
+  const { data: questionnaireData } = useQuery({
+    queryKey: ["questionnaire", questionnaireSlug],
+    queryFn: query(questionnaireApi.detail, {
+      pathParams: { id: questionnaireSlug! },
+    }),
+    enabled: !!questionnaireSlug,
+  });
+
+  useEffect(() => {
+    if (questionnaireData?.title) {
+      setSelectedQuestionnaireTitle(questionnaireData.title);
+    }
+  }, [questionnaireData]);
```

Comment:
do we really need this useEffect and state actually? 🤔 

can't we simply use `questionnaireData?.title` directly wherever `selectedQuestionnaireTitle` is being used? 🤔